File: frontend/src/types/factura.ts
import { ReservaNested } from './reserva';

export interface Factura {
  id: number;
  reserva_id: number;
  total: number;
  estado: string;
  metodo_pago?: string;
  reserva?: ReservaNested;
  created_at: string;
  updated_at?: string;
}

export interface FacturaCreate {
  reserva_id: number;
  total: number;
  estado?: string;
  metodo_pago?: string;
}

export interface FacturaUpdate {
  total?: number;
  estado?: string;
  metodo_pago?: string;
}

File: backend/app/schemas/factura.py
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class ReservaNested(BaseModel):
    id: int
    fecha_entrada: date
    fecha_salida: date

    model_config = ConfigDict(from_attributes=True)


class FacturaBase(BaseModel):
    reserva_id: int
    total: float
    estado: str = "pendiente"
    metodo_pago: Optional[str] = None


class FacturaCreate(FacturaBase):
    pass


class FacturaUpdate(BaseModel):
    total: Optional[float] = None
    estado: Optional[str] = None
    metodo_pago: Optional[str] = None


class FacturaResponse(FacturaBase):
    id: int
    reserva: Optional[ReservaNested] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)
